util/reggen/systemrdl/udp.py

```diff
@@ -0,0 +1,47 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+
+from pathlib import Path
+from typing import Any
+from systemrdl.udp import UDPDefinition
+from systemrdl import RDLCompiler  # type: ignore[attr-defined]
+from systemrdl.node import Node
+from systemrdl.component import Field, Reg
+
+
+class UDPBoolean(UDPDefinition):
+    valid_type = bool
+    default_assignment = False
+
+    def validate(self, node: Node, value: Any) -> None:
+        if not isinstance(value, bool):
+            self.msg.error("The type of {self.name} must be an bool.", self.get_src_ref(node))
+
+
+class Mubi(UDPBoolean):
+    name = "mubi"
+    valid_components = {Field}
+
+
+class Hwre(UDPBoolean):
+    name = "Hwre"
+    valid_components = {Reg}
+
+
+class Shadowed(UDPBoolean):
```

Comment:
Seems fine. There's is no standard property for this, only vendor-defined ones. It's too bad the SystemRDL WG is inactive.